pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -336,24 +348,69 @@ pub mod pallet {
 			)
 		}
 
-		#[pallet::call_index(3)]
+		#[pallet::call_index(2)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
 		pub fn reset_manager(
 			origin: OriginFor<T>,
 			pool_id: T::PoolId,
 			manager: Option<AccountIdOf<T>>,
 		) -> DispatchResult {
 			let who = T::DefaultOrigin::ensure_origin(origin)?;
-			Pools::<T>::try_mutate(pool_id, |maybe_entry| -> DispatchResult {
+			Pools::<T>::try_mutate(&pool_id, |maybe_entry| -> DispatchResult {
 				let entry = maybe_entry.as_mut().ok_or(Error::<T>::PoolUnknown)?;
 				ensure!(entry.is_manager(&who), Error::<T>::NoPermission);
-				entry.manager = manager;
+				entry.manager = manager.clone();
+
+				Self::deposit_event(Event::ManagerUpdate {
+					id: pool_id.clone(),
+					manager,
+				});
+
 				Ok(())
 			})
 		}
 
+		#[pallet::call_index(3)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn set_lock(origin: OriginFor<T>, pool_id: T::PoolId, lock: Locks) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
```

Comment:
we prob have to have another look at the origins, now some of the management origins are PoolCreateOrigins, others are DefaultOrigins?